test/Interop/Cxx/enum/Inputs/anonymous-with-swift-name.h

```diff
@@ -0,0 +1,22 @@
+#ifndef TEST_INTEROP_CXX_ENUM_INPUTS_ANONYMOUS_WITH_SWIFT_NAME_H
+#define TEST_INTEROP_CXX_ENUM_INPUTS_ANONYMOUS_WITH_SWIFT_NAME_H
+
+#define SOME_OPTIONS(_type, _name) __attribute__((swift_name("_TYPEDEF_" #_name))) _type _name; enum __attribute__((swift_name(#_name))) __attribute__((flag_enum,enum_extensibility(open))) : _type
```

Comment:
I don't think this pattern will work for us, if only because users may want to use `swift_name()` to give their option sets custom names:

```cpp
typedef SOME_OPTIONS(unsigned, SOColorMask) {
     kSOColorMaskRed = (1 << 1),
     kSOColorMaskGreen = (1 << 2),
     kSOColorMaskBlue = (1 << 3),
     kSOColorMaskAll = ~0U
 } __attribute__((swift_name("SOColorOptions")));
// Oops! Now the name relationship between typedef and enum is totally busted.
```

However, I think your general approach of modifying the macro to link the typedef and enum together by name is a good one. It just needs to be done by the clang name, not the Swift name. For instance, suppose we extended `__attribute__((flag_enum))` to allow an optional (C++) type name; we could then link the two by their clang names, leaving the `swift_name` attribute to do its current job:

```cpp
#define SOME_OPTIONS(_type, _name) _type _name; enum __attribute__((flag_enum(_name),enum_extensibility(open))) : _type
```

(You could instead introduce a new attribute—say, `anonymous_typedef(TypeName)`—to represent this link; it wouldn't be much different.)

Another way to do it might be to require the typedef to be used as the underlying type of the anonymous enum, like so:

```
#define SOME_OPTIONS(_type, _name) _type _name; enum __attribute__((flag_enum,enum_extensibility(open))) : _name
// Note that we're using `: _name` instead of `: _type`.
```
I believe these are equivalent to C++, but we can give them different meanings to Swift.

(There *is* a danger that someone might accidentally turn, say, `NSInteger` into an options enum by using the old macro with the new compiler. Perhaps you'd need to also mark the typedef with `flag_enum`—this might require an extension to clang so that it would allow the attribute to be applied to typedefs.)

In any case, we're going to want attributes applied to the enum to also be respected when we import the typedef. Actually, what we probably want is for `ImportDecl.cpp:canSkipOverTypedef()` to tell us to import the enum instead, but also to fall back to the typedef's name if there isn't a `swift_name` attribute.